src/discovery/discoveryLogic.ts

```diff
@@ -1,90 +1,247 @@
-import { NamedNode, Namespace, LiveStore } from "rdflib";
+import { NamedNode, Namespace, LiveStore, sym, st } from "rdflib";
+import * as debug from '../util/debug'
+// import { getContainerMembers } from '../util/UtilityLogic'
+import { solidLogicSingleton } from "../logic/solidLogicSingleton"
 
-type TypeIndex = {  label: string, index: NamedNode, agent: NamedNode } ;
+const {  getContainerMembers, authn } = solidLogicSingleton
+const { currentUser } = authn
+
+type TypeIndexScope = { label: string, index: NamedNode, agent: NamedNode } ;
+type ScopedApp = { instance: NamedNode, scope: TypeIndexScope }
 
 const ns ={
+  rdf:   Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#'),
   solid: Namespace('http://www.w3.org/ns/solid/terms#'),
   space: Namespace('http://www.w3.org/ns/pim/space#')
 }
 
-export async function loadProfile(store: LiveStore, user) {
+/** Create a resource if it really does not exist
+ *  Be absolutely sure something does not exist before creating a new empty file
+ * as otherwise existing could  be deleted.
+ * @param doc {NamedNode} - The resource
+ */
+export async function loadOrCreateIfNotExists (store: LiveStore, doc: NamedNode) {
+    let response
+    try {
+      response = await store.fetcher.load(doc)
+    } catch (err) {
+        if (err.response.status === 404) {
+          debug.log('createIfNotExists doc does NOT exist, will create: ' + doc)
+          try {
+            store.fetcher.webOperation('PUT', doc.uri, {data: '', contentType: 'text/turtle'})
+          } catch (err) {
+            const msg = 'createIfNotExists doc FAILED: ' + doc + ': ' + err
+            debug.log(msg)
+            throw new Error(msg)
+          }
+          delete store.fetcher.requested[doc.uri] // delete cached 404 error
```

Comment:
I did not know about this call. I haven't seen it so far. 
Question: don't we have to load the newly created resource into the cache anymore?
I expect this mostly from the name of the function "load". Of course, it depends on what 'load' means -> load in the cache presumably. And it also depends if the creation automatically loads in cache beforehand. 
So what is the truth?